tests/testthat/test-disability.R

```diff
@@ -0,0 +1,74 @@
+test_that("global and component-wise outputs align", {
+
+  ## Check 1: Make sure that global & component-wise flags don't contradict each other
+  # create dummy data
+  set.seed(1)
+  ipdiagnosis <- dummy_diag(nid = 100, nrow = 400)
+  erdiagnosis <- dummy_diag(nid = 80, nrow = 200, ipdiagnosis = FALSE)
+  cohort <- data.table(genc_id = c(unique(ipdiagnosis$genc_id), 888, 999)) # add some genc_ids with no diagnosis entries
+
+  check1_no_cat <- disability(cohort, ipdiag = ipdiagnosis, erdiag = erdiagnosis, component_wise = FALSE)
+  check1_cat <- disability(cohort, ipdiag = ipdiagnosis, erdiag = erdiagnosis, component_wise = TRUE)
+
+  ## All genc_ids where global disability flag = TRUE should exist in component-wise output (and v.v.)
+  expect_true(all(check1_no_cat[disability == TRUE]$genc_id %in% check1_cat$genc_id == TRUE))
+  expect_true(all(check1_no_cat[disability == FALSE]$genc_id %in% check1_cat$genc_id == FALSE))
+  expect_true(all(check1_cat$genc_id %in% check1_no_cat[disability == TRUE]$genc_id == TRUE))
+
+  ## If no entry in diagnosis table, disability should be NA in global disability output
+  expect_equal(unique(check1_no_cat[genc_id == 888 | genc_id == 999, disability]), NA)
+
+})
+
+
+test_that("returned with disability = TRUE", {
+
+  ## Check 2: Unit test for some diagnosis codes that should have disability = TRUE
+  check2 <- disability(cohort = data.table(genc_id = c(1,2,3,4,5)),
+                       ipdiag = data.table(genc_id = c(1,1,2,2,2,3,4,5,5,5,5,5),
+                                           diagnosis_code = c('F7001', 'Q8723','Q66', 'E2204', 'M23836', 'G60', 'G242','S029', 'Z998', 'H90132', 'Q150', 'E10352')),
```

Comment:
I generally agree that we should use the dummy functions as much as possible. The issue I ran into was that some diagnoses I wanted to test weren't created with the dummy function. E.g., running this `ipdiag <- dummy_diag(nid=10, nrow=100, ipdiagnosis=T, pattern ="^F700")` results in an error, probably because F700 is a Canadian code, so I'd need to connect to the DB to create this code. It seemed easier to just make up fake examples of things I wanted to test than check whether all the different scenarios I wanted to test are covered by the randomly generated dummy data. However, if you feel strongly about this, I can pick other diagnosis prefixes and make sure the dummy data covers all the scenarios I want to test.

Reply:
Yes if you want to sample from Canadian code right now a db connection is needed - could be a future enhancement.  Generally I would think if a function picks up e.g. "^F840" it would work the same for F700. So if an alternative can achieve the same goal for the test I'd try to use the dummy fun. If there is a specific reason that we want to test specifically for F700 I don't feel strongly about diverging from using dummy fun.

Reply:
Yes, I think the main problem is that a lot of code prefixes only return the exact same code with the `dummy_diag` function because of that (e.g., "^F840" only returns exactly codes = "F840"), so I had to hand-pick a few other codes to make sure I'm also getting some that are actually children codes of the ones in the mapping file. This can easily be missed when using the dummy functions, so just something to keep in mind (plus, in this case, I wanted to cover all the different categories so had to generate an unnecessarily large dummy input to achieve that). Anyway, I did change the test as you suggested and I think it's fine now.

Reply:
All make sense! Pros and cons of sampling randomness. And sometimes the function doesn't return child level codes is due to that the child level codes do not actually exist (no further sublevels). We can keep this as a future enhancement - to allow greater user control on the codes returned and potentially allow sampling from icd-10-ca without db connection.